### PR TITLE
Document the fork build security procedures

### PR DESCRIPTION
**Pull request by techcobweb**
Signed-off-by: Mike Cobbett <77053+techcobweb@users.noreply.github.com>
- Some docs on how to contribute and what approvals are needed.

### CODE REVIEW

**Comment by galasa-team** [comment]
Build successful

**Review by jadecarino — APPROVED** [review]

**Comment by galasa-team** [comment]
Build successful

**Comment by galasa-team** [comment]
Build successful

File: build-images/github-verify/cmd/main.go [hunk 1]
```diff
@@ -21,6 +24,21 @@ var client = http.Client{
 	Timeout: 30 * time.Second,
 }
 
+// This program is called either because:
+//   - Some pull request has been opened/re-opened/synchronized.
+//     (in which case the userId passed is the person who did the push/open/re-open)
+//   - Some pull request has been reviewed, and the comment "Approved for building"
+//     has been submitted as a review comment.
+//     (in which case the userId passed is hte person who did the review)
```
Comment (eamansour) [inline build-images/github-verify/cmd/main.go:32]: Tiny typo - "hte"

File: docs/pull-request-build-authentication.md [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+
+1. When they open the PR, github will call a Web Hook Receiver program. It doesn't do anything with the web hook event, merely returning HTTP 200 (OK), so that Github thinks it delivered the webhook event.
+2. At some point in time after that happened, a cron job in Kubernetes will fire (every minute).
+It calls a [web hook monitor](../build-images/github-webhook-monitor) program.
+3. The web hook monitor calls out to Github, through the IBM firewall, and interrogates Github for information about 
+recent web hook events it has sent out for the galasa-dev organisation.
+For each webhook event it finds in the Github events history, it sends the exact webhook payload to Tekton.
+When there are no more events to process in Github, the monitor saves a 'bookmark' of the latest event it has processed 
+and exits.
+When the monitor next wakes up, it reads the bookmark, and knows where to start 'replaying' the github events from, therebye avoiding issuing duplicate hook event copies to Tekton.
```
Comment (eamansour) [inline docs/pull-request-build-authentication.md:32]: Typo - "therebye"

File: docs/pull-request-build-authentication.md [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+4. The event indicating the PR was opened is sent to Tekton. The user id in the event is the id of the non-trusted developer.
+6. The `pr-builds` Tekton listener fires, and processes the 'PR open' event. The user id in the even is the id of the non-trusted developer still.
```
Comment (eamansour) [inline docs/pull-request-build-authentication.md:34]: Missing "t" in "event"

**Review by eamansour — APPROVED** [review]
Looks good! Just spotted a few small typos worth fixing but overall great documentation.

**Comment by galasa-team** [comment]
Build successful

**Review by eamansour — APPROVED** [review]

**Pull request merged**